Corona-Warn-App/src/main/java/de/rki/coronawarnapp/contactdiary/ui/edit/ContactDiaryEditLocationsFragment.kt

```diff
@@ -0,0 +1,107 @@
+import de.rki.coronawarnapp.contactdiary.ui.edit.ContactDiaryEditLocationsViewModel.NavigationEvent.ShowLocationDetailSheet
+import de.rki.coronawarnapp.databinding.ContactDiaryEditLocationsFragmentBinding
+import de.rki.coronawarnapp.util.DialogHelper
+import de.rki.coronawarnapp.util.di.AutoInject
+import de.rki.coronawarnapp.util.ui.observe2
+import de.rki.coronawarnapp.util.ui.popBackStack
+import de.rki.coronawarnapp.util.ui.viewBindingLazy
+import de.rki.coronawarnapp.util.viewmodel.CWAViewModelFactoryProvider
+import de.rki.coronawarnapp.util.viewmodel.cwaViewModels
+import javax.inject.Inject
+
+class ContactDiaryEditLocationsFragment : Fragment(R.layout.contact_diary_edit_locations_fragment), AutoInject {
+
+    @Inject lateinit var viewModelFactory: CWAViewModelFactoryProvider.Factory
+    private val viewModel: ContactDiaryEditLocationsViewModel by cwaViewModels { viewModelFactory }
+    private val binding: ContactDiaryEditLocationsFragmentBinding by viewBindingLazy()
+
+    private val locationList : MutableList<ContactDiaryLocation> = mutableListOf()
+    private val listAdapter = ListAdapter()
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        binding.viewModel = viewModel
+
+        setupRecyclerView()
+
+        binding.toolbar.setNavigationOnClickListener {
+            popBackStack()
+        }
+
+        viewModel.locationsLiveData.observe2(this) {
+            locationList.clear()
+            locationList.addAll(it)
+            listAdapter.notifyDataSetChanged()
+        }
+
+        viewModel.navigationEvent.observe2(this) {
+
+            when(it) {
+                ShowDeletionConfirmationDialog ->  DialogHelper.showDialog(deleteAllLocationsConfirmationDialog)
+                is ShowLocationDetailSheet -> {
+                    // todo
+                }
+            }
+        }
+    }
+
+    override fun onResume() {
+        super.onResume()
+        binding.contentContainer.sendAccessibilityEvent(AccessibilityEvent.TYPE_ANNOUNCEMENT)
+    }
+
+    private fun setupRecyclerView() {
+        binding.locationsRecyclerView.adapter = listAdapter
+    }
+
+    private val deleteAllLocationsConfirmationDialog by lazy {
+        DialogHelper.DialogInstance(
+            requireActivity(),
+            R.string.contact_diary_delete_locations_title,
+            R.string.contact_diary_delete_locations_message,
+            R.string.contact_diary_delete_button_positive,
+            R.string.contact_diary_delete_button_negative,
+            positiveButtonFunction = {
+                viewModel.onDeleteAllConfirmedClick()
+            }
+        )
+    }
+
+    inner class ListAdapter : RecyclerView.Adapter<ListAdapter.ViewHolder>() {
+
+        inner class ViewHolder(listItemView: View) : RecyclerView.ViewHolder(listItemView) {
+            val nameTextView = itemView.findViewById<TextView>(R.id.name)
```

Comment:
could use view binding here but is really up to you

Reply:
for one view it's easier this way